learning/tour-of-beam/frontend/lib/pages/tour/widgets/unit_content.dart

```diff
@@ -183,25 +154,114 @@ class _ChallengeButtons extends StatelessWidget {
     final hints = unitContent.hints;
 
     return Row(
+      crossAxisAlignment: CrossAxisAlignment.start,
       mainAxisAlignment: MainAxisAlignment.end,
       children: [
-        if (unitContent.isChallenge)
+        _SnippetTypeSwitcher(
+          tourNotifier: tourNotifier,
+          unitContent: unitContent,
+        ),
+        if (hints.isNotEmpty)
           Padding(
             padding: _buttonPadding,
             child: HintsWidget(
               hints: hints,
             ),
           ),
-        if (tourNotifier.doesCurrentUnitHaveSolution)
-          Padding(
-            padding: _buttonPadding,
-            child: SolutionButton(tourNotifier: tourNotifier),
-          ),
       ],
     );
   }
 }
 
+class _SnippetTypeSwitcher extends StatelessWidget {
+  final TourNotifier tourNotifier;
+  final UnitContentModel unitContent;
+
+  const _SnippetTypeSwitcher({
+    required this.tourNotifier,
+    required this.unitContent,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    // TODO(nausharipov): styling
+    return AnimatedBuilder(
+      animation: tourNotifier,
+      builder: (context, child) => Padding(
+        padding: const EdgeInsets.all(BeamSizes.size10),
+        child: Row(
+          children: [
+            if (tourNotifier.hasSolution)
+              _SnippetTypeButton(
+                buttonSnippetType: SnippetType.solution,
+                tourNotifier: tourNotifier,
+                label: 'pages.tour.solution'.tr(),
+              ),
+            if (tourNotifier.hasSolution || tourNotifier.hasSavedSnippet)
+              _SnippetTypeButton(
+                buttonSnippetType: SnippetType.original,
+                tourNotifier: tourNotifier,
+                label: unitContent.isChallenge
+                    ? 'pages.tour.assignment'.tr()
+                    : 'pages.tour.example'.tr(),
+              ),
+            if (tourNotifier.hasSavedSnippet)
+              _SnippetTypeButton(
+                buttonSnippetType: SnippetType.saved,
+                tourNotifier: tourNotifier,
+                label: tourNotifier.saveCodeStatus == SaveCodeStatus.saving
+                    ? 'pages.tour.saving'.tr()
+                    : 'pages.tour.myCode'.tr(),
+              ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+class _SnippetTypeButton extends StatelessWidget {
+  final SnippetType buttonSnippetType;
+  final String label;
+  final TourNotifier tourNotifier;
```

Comment:
```suggestion
  final SnippetType groupValue;
  final String title;
  final SnippetType value;
```

Reply:
I didn't add `groupValue`, because it causes unnecessary duplication of `tourNotifier.snippetType`.